### ISSUE TITLE
Installation on macOS High Sierra

### ISSUE THREAD

**Issue by feschmidt** [gsteele13/spyview#3]
I attempted to install spyview on macOS High Sierra from source the following way:
```
cd spyview/source
autoreconf -i
./configure
make
```
`autoreconf -i` was required to create the `configure` bash.
After the above, I got an error about missing `constexpr` definitions in `./spyview/class.H`. This could be solved by changing the indicated instances of `const` to `constexpr` as indicated by the error message (thanks to @yausern)
`make` then gave the following error:
```
./mypam.h:5:10: fatal error: 'pam.h' file not found
#include <pam.h>
```
repeating `autoreconf -i` and `./configure` doesn't help. However, `pam.h` is definitely on system installed via `brew install netpbm`. My workaround was to copy `pam.h` and `pm.h` (about which `make` complains right after) into `./spyview/`

Next error message after `make`:
```
spyview.C:687:16: error: no matching literal operator for call to
      'operator""_STF' with arguments of types 'const char *' and
      'unsigned long', and no matching literal operator template
  info("Found "_STF" files\n", filenames.size());
               ^
spyview.C:696:17: error: no matching literal operator for call to
      'operator""_STF' with arguments of types 'const char *' and
      'unsigned long', and no matching literal operator template
  info("Loaded "_STF" color maps from %s.\n",cmapfiles.size(),share_path...
                ^
spyview.C:701:21: error: no matching literal operator for call to
      'operator""_STF' with arguments of types 'const char *' and
      'unsigned long', and no matching literal operator template
      info("Loaded "_STF" color maps from %s.\n",cmapfiles.size(),user_p...
                    ^
```

I have no idea how to go on from here

**Comment by feschmidt** [comment]
OK, this can be fixed by converting the three respective lines into 
```
info("Found ",_STF," files\n", filenames.size());
```
i.e. adding commata in between the `str` and `_STF`.
Next error then:
```
ImageData.C:2848:9: error: cannot initialize a variable of type 'char *' with an
      rvalue of type 'const char *'
  char *p = strstr(comments, "Loop 2");
        ^   ~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by feschmidt** [comment]
fixed by converting to
```
const char *p = strstr(comments, "Loop 2");
```
Next error:
```
ImageWindow_Fitting.C:1031:42: error: no matching literal operator for call to
      'operator""_STF' with arguments of types 'const char *' and
      'unsigned long', and no matching literal operator template
      snprintf(ubuf,sizeof(ubuf),"Undo ("_STF")",undoStack.size());
                                         ^
ImageWindow_Fitting.C:1043:42: error: no matching literal operator for call to
      'operator""_STF' with arguments of types 'const char *' and
      'unsigned long', and no matching literal operator template
      snprintf(rbuf,sizeof(rbuf),"Redo ("_STF")",redoStack.size());
                                         ^
```
Also inserted commata here.
```
In file included from spypal.C:10:
./bisector.H:12:23: error: in-class initializer for static data member of type
      'const double' requires 'constexpr' specifier [-Wstatic-float-init]
  static const double factor=0.5;
                      ^      ~~~
./bisector.H:12:3: note: add 'constexpr'
  static const double factor=0.5;
  ^
  constexpr 
./bisector.H:72:23: error: in-class initializer for static data member of type
      'const double' requires 'constexpr' specifier [-Wstatic-float-init]
  static const double factor=0.5;
                      ^      ~~~
./bisector.H:72:3: note: add 'constexpr'
  static const double factor=0.5;
  ^
  constexpr 
```
Converted to `constexpr` as before.
```
gp_to_mat.C:199:36: error: no matching literal operator for call to
      'operator""_STF' with arguments of types 'const char *' and
      'unsigned long', and no matching literal operator template
  fprintf(stdout,"#pre-cull size: "_STF","_STF"\n",xs.size(),ys.size());
                                   ^
```
Inserted commata.

This fixed it!!
Completed installation with `sudo make install` and now able to open with `spyview` from terminal

@gsteele13 can you add me as collaborator so I can push the changes?
One thing though is that it appears to be quite slow on my system. Maybe some of the above changes are not good practice?

**Comment by feschmidt** [comment]
Fixed in [`a97bb61`](https://github.com/gsteele13/spyview/commit/a97bb61172a451b6f7d9afc61b41bae007d1757a)
